test/unit/org/apache/cassandra/cql3/PasswordObfuscatorTest.java

```diff
@@ -0,0 +1,160 @@
+    {
+        assertEquals(format("ALTER ROLE role1 with PASSWORD = '%s'", PasswordObfuscator.OBFUSCATION_TOKEN),
+                     obfuscator.obfuscate("ALTER ROLE role1 with PASSWORD = '123'"));
+    }
+
+    @Test
+    public void testAlterRoleWithPasswordNoSpace()
+    {
+        assertEquals(format("ALTER ROLE role1 with PASSWORD='%s'", PasswordObfuscator.OBFUSCATION_TOKEN),
+                     obfuscator.obfuscate("ALTER ROLE role1 with PASSWORD='123'"));
+    }
+
+    @Test
+    public void testAlterRoleWithPasswordNoImmediateSpace()
+    {
+        assertEquals(format("ALTER ROLE role1 with PASSWORD= '%s'", PasswordObfuscator.OBFUSCATION_TOKEN),
+                     obfuscator.obfuscate("ALTER ROLE role1 with PASSWORD= '123'"));
+    }
+
+    @Test
+    public void testAlterRoleWithoutPassword()
+    {
+        assertEquals("ALTER ROLE role1", obfuscator.obfuscate("ALTER ROLE role1"));
+    }
+
+    @Test
+    public void testCreateUserWithPassword()
+    {
+        assertEquals(format("CREATE USER user1 with PASSWORD '%s'", PasswordObfuscator.OBFUSCATION_TOKEN),
+                     obfuscator.obfuscate("CREATE USER user1 with PASSWORD '123'"));
+    }
+
+    @Test
+    public void testCreateUserWithoutPassword()
+    {
+        assertEquals("CREATE USER user1", obfuscator.obfuscate("CREATE USER user1"));
+    }
+
+    @Test
+    public void testAlterUserWithPassword()
+    {
+        assertEquals(format("ALTER USER user1 with PASSWORD '%s'", PasswordObfuscator.OBFUSCATION_TOKEN),
+                     obfuscator.obfuscate("ALTER USER user1 with PASSWORD '123'"));
+    }
+
+    @Test
+    public void testAlterUserWithPasswordMixedCase()
+    {
+        assertEquals(format("ALTER USER user1 with paSSwoRd '%s'", PasswordObfuscator.OBFUSCATION_TOKEN),
+                     obfuscator.obfuscate("ALTER USER user1 with paSSwoRd '123'"));
+    }
+
+    @Test
+    public void testAlterUserWithPasswordWithNewLine()
+    {
+        assertEquals(format("ALTER USER user1 with PASSWORD\n'%s'", PasswordObfuscator.OBFUSCATION_TOKEN),
+                     obfuscator.obfuscate("ALTER USER user1 with PASSWORD\n'123'"));
+    }
+
+    @Test
+    public void testPasswordWithNewLinesObfuscation()
+    {
+        assertEquals(String.format("CREATE USER user1 with PASSWORD '%s'", PasswordObfuscator.OBFUSCATION_TOKEN),
+                     obfuscator.obfuscate("CREATE USER user1 with PASSWORD 'a\nb'"));
+    }
+
+    @Test
+    public void testEmptyPasswordObfuscation()
+    {
+        assertEquals(String.format("CREATE USER user1 with PASSWORD '%s'", PasswordObfuscator.OBFUSCATION_TOKEN),
+                     obfuscator.obfuscate("CREATE USER user1 with PASSWORD ''"));
+    }
+
+    @Test
+    public void testPasswordWithSpaces()
+    {
+        assertEquals(String.format("CREATE USER user1 with PASSWORD '%s'", PasswordObfuscator.OBFUSCATION_TOKEN),
+                     obfuscator.obfuscate("CREATE USER user1 with PASSWORD 'p a ss wor d'"));
```

Comment:
Theoretically we can add a test for `CREATE ROLE user1 WITH PASSWORD = 'password_a' AND LOGIN = true AND ACCESS TO DATACENTERS {'DC1', 'DC3'};` - the idea is to have two variables with quotes and ensure that the regex works in "greedy manner" and obfuscates `password_a` instead of `password_a' AND LOGIN = true AND ACCESS TO DATACENTERS {'DC1', 'DC3`. I'm pretty sure the current logic works properly, so it is up to you whether to add such a test.

Reply:
resolving since we moved to a simpler logic from using regex